### PR TITLE
[mail] Fix imap protocol setup error

### PR DESCRIPTION
**Pull request by 9037568**
While researching [this problem](https://community.openhab.org/t/problem-refreshing-imap-pop3/95637/6) in the community forum, I found this gremlin.

The protocol setup fails due to [this line](https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java#L74) in the handler, which does nothing:

     protocol.concat("s");

So I fixed that and added some actual debug output.

Signed-off-by: 9037568 <namraccr@gmail.com>

### CODE REVIEW

File: bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/config/POP3IMAPConfig.java [hunk 1]
```diff
@@ -23,4 +23,15 @@
 @NonNullByDefault
 public class POP3IMAPConfig extends BaseConfig {
     public int refresh = 60;
+    
+    public String debugPrint() {
+        StringBuffer buffer = new StringBuffer();
```
Comment (cpmeister) [inline bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/config/POP3IMAPConfig.java:28]:
```suggestion
        StringBuilder buffer = new StringBuilder();
```

File: bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java [hunk 2]
```diff
@@ -94,6 +94,7 @@ public void initialize() {
             }
         }
 
+        logger.debug("Configuration details:\r\n{}", config.debugPrint());
```
Comment (cpmeister) [inline bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java:97]:
```suggestion
        if(logger.isDebugEnabled()){
            logger.debug("Configuration details:\r\n{}", config.debugPrint());
        }
```

File: bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java [hunk 3]
```diff
@@ -71,7 +71,7 @@ public void initialize() {
         protocol = baseProtocol;
 
         if (config.security == ServerSecurity.SSL) {
-            protocol.concat("s");
+            protocol = protocol.concat("s");
```
Comment (J-N-K) [inline bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java:74]: This is the only useful change here. Please remove all the log-bloating debugging.

File: bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/config/POP3IMAPConfig.java (hunk 1 above)
Comment (9037568) [reply bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/config/POP3IMAPConfig.java:28]: Easy enough, but please explain why you think this is needed.

File: bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java (hunk 3 above)
Comment (9037568) [reply bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java:74]: Not going to happen. Debug logging is needed and useful.

File: bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java (hunk 2 above)
Comment (9037568) [reply bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java:97]: This method is very low cost. Why do you feel the if conditional is needed here?

File: bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java (hunk 3 above)
Comment (J-N-K) [reply bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/POP3IMAPHandler.java:74]: As a Codeowner I prefer to keep the badly written style we had before.

**Pull request closed without merging**